File: FizzBuzz/prime_word.rb
#!/usr/bin/env ruby
# encoding: utf-8

class PrimeWord

  def initialize( prime, word )
    @prime = prime
    @word  = word
  end

  def say( value )
    value % @prime == 0 ? @word : ''
  end

end

if __FILE__ == $0
  instance = PrimeWord.new( 3, 'Fizz' )
  puts instance.say( 15 )
end

File: FizzBuzz/PrimeWord.java
// Compile command: javac PrimeWord.java

public class PrimeWord {

  private final Integer prime;
  private final String  word ;

  public PrimeWord( final Integer prime, final String word ) {
    this.prime = prime;
    this.word  = word ;
  }

  public String say( final Integer value ) {
    return ( value % prime == 0 ) ? word : "";
  }

}